src/Rules/Classes/EnumSanityRule.php

```diff
@@ -122,6 +124,42 @@ public function processNode(Node $node, Scope $scope): array
 			}
 		}
 
+		if ($node->scalarType !== null) {
+
+			$enumCases = [];
+			foreach ($node->stmts as $stmt) {
+				if (!($stmt instanceof Node\Stmt\EnumCase && ($stmt->expr instanceof Node\Scalar\LNumber || $stmt->expr instanceof Node\Scalar\String_))) {
```

Comment:
You could check more things here. Like:

1) Wrong case value type against the type declared in `$node->scalarType`.
2) Missing case value in backed enum, and extra case value in case of non-backed enum.

All three cases here should fail: https://phpstan.org/r/3ca06994-1a6e-4686-a57c-6083c6bdb34e

Reply:
Thanks for the suggestions, I'll implement them.

Reply:
@ondrejmirtes Done. 3 tests are failing but they seem like unrelated to mine changes.